pkg/trace/api/otlp.go

```diff
@@ -288,6 +311,34 @@ func (o *OTLPReceiver) ReceiveResourceSpans(ctx context.Context, rspans ptrace.R
 	return src
 }
 
+// createChunks creates a set of pb.TraceChunk's based on two maps:
+// - a map from trace ID to the spans sharing that trace ID
+// - a map of user-set sampling priorities by trace ID, if set
+func (o *OTLPReceiver) createChunks(tracesByID map[uint64]pb.Trace, prioritiesByID map[uint64]sampler.SamplingPriority) []*pb.TraceChunk {
+	traceChunks := make([]*pb.TraceChunk, 0, len(tracesByID))
+	for k, spans := range tracesByID {
+		if len(spans) == 0 {
+			continue
+		}
+		rate := strconv.FormatFloat(o.samplingRate(), 'f', 2, 64)
+		chunk := &pb.TraceChunk{
+			Tags:  map[string]string{"_dd.otlp_sr": rate},
+			Spans: spans,
+		}
+		if p, ok := prioritiesByID[k]; ok {
+			// a manual decision has been made by the user
+			chunk.Priority = int32(p)
+			traceutil.SetMeta(spans[0], "_dd.p.dm", "-4")
+		} else {
+			// we use the probabilistic sampler to decide
+			chunk.Priority = int32(o.sample(k))
+			traceutil.SetMeta(spans[0], "_dd.p.dm", "-9")
```

Comment:
Are we tracking `_dd.p.dm` values somewhere? How did you come up with `9`?

Reply:
Answered in private. Yes.